web/helpers/validations.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Community.Validations do
+  import Ecto.Changeset
+
+  def validate_url_format(changeset, field) do
+    format = ~r/http(s)?.*/
+    message = "must start with http(s)"
+
+    case get_field(changeset, field) do
+      nil ->
+        changeset
+      string when byte_size(string) > 0 ->
+        validate_format(changeset, field, format, message: message)
+      string ->
+        changeset
+    end
```

Comment:
I think this could be simplified with

```elixir
if get_field(changeset, field) in [nil, ""] do
  changeset
else
  validate_format(changeset, field, format, message: message)
end
```